ctmds1/power_setup.py

```diff
@@ -110,20 +64,48 @@ class Demand(str, Enum):
 }
 
 
+def allocate_power(power_sources, demand_mwh):
+    sorted_sources = sorted(power_sources.items(), key=lambda x: x[1]["cost"])
+
+    remaining_demand = demand_mwh
+    total_cost = 0
+    allocation = []
+
+    for source, data in sorted_sources:
+        if remaining_demand <= 0:
+            break
+        supply = min(data["capacity"], remaining_demand)
+        cost_for_source = supply * data["cost"]
+        allocation.append(
+            {"source": source.value, "supply": supply, "cost": data["cost"]}
+        )
+        total_cost += cost_for_source
```

Comment:
Watch out, there's something counter intuitive here! 

The total cost is not (wind supply * the cost of wind + gas supply * the cost of gas) for example...  For any given amount of demand, every participant from the stack is paid at the HIGHEST price. So the total price is (wind supply + gas supply) * cost of gas 

Does this make sense?

In other words, even if renewables provide 90% of the demand for the hour and gas is only needed for the last 10%, EVERYONE gets paid at the gas price point. And if coal kicks in even for a brief time, everyone (including renewables) will get paid at the coal price for that hour.